src/lpython/parser/parser.cpp

```diff
@@ -133,5 +132,15 @@ Result<LPython::AST::ast_t*> parse_python_file(Allocator &al,
     return ast;
 }
 
+Result<LPython::AST::ast_t*> parse_python_file(Allocator &al,
+        const std::string &runtime_library_dir,
+        const std::string &infile,
+        diag::Diagnostics &diagnostics,
+        uint32_t prev_loc,
+        [[maybe_unused]] bool new_parser) {
+    std::string input_source_code = read_file(infile);
+    return parse_python_source(al, runtime_library_dir, input_source_code, diagnostics, prev_loc, new_parser);
+}
```

Comment:
Instead, refactor the code to not read in `parse_python_file` function in `parser.cpp`.